Foundation/NSData.swift

```diff
@@ -145,10 +145,20 @@ public class NSData : NSObject, NSCopying, NSMutableCopying, NSSecureCoding {
         return true
     }
     
-    override public var description: String {
-        get {
-            return "Fixme"
+    private var byteDescription: String {
+        var s = ""
+        let buffer = UnsafePointer<UInt8>(bytes)
+        for i in (0..<self.length) {
+            if i > 0 && i % 4 == 0 { s += " " }
+            var byteStr = String(buffer[i], radix: 16, uppercase: false)
+            if byteStr.characters.count == 1 { byteStr = "0\(byteStr)" }
```

Comment:
from my understanding description does a full hex dump and debugDescription does a truncated version.